### PR TITLE
Moves damage deflection code from doors to machines and gives it to the stacking machine

### PR DESCRIPTION
**Pull request by lilacrhododendron**
:cl: 
code: Damage deflection is handled at the machine level instead of the door level now, so it can be applied to any machine.
balance: The stacking machine now has damage deflection, so it can't be destroyed by things like pickaxes.
/:cl:

There's a conundrum that has occurred more than I'd like it to wherein people who get gulagged proceed to destroy one of the stacking machines either on the gulag itself or in the shuttle and basically destroy the ability for the gulag to function whatsoever. Since there's no way for a player to rebuild this machine, it becomes a big pain in the butt and everyone else who is with them on the gulag is stranded there. My thought was to simply give it damage deflection like airlocks, but then I discovered that was airlock exclusive, so I've moved it to machines. This could be applied to a number of other machines. The stacking machine is the most obvious because of the limited access to tools. If we wanted to be really anti-fun (or more fun, depending on how you look at it) it could also be applied to something like the permabrig machines. Really, any situation where you need a machine that is also resistant to weak weapons from people with a lot of time. 21 is a placeholder value that I chose based on the one that airlocks have. It doesn't have to be that number, it just needs to be higher than pickaxe damage (15).

### CODE REVIEW

File: code/game/machinery/_machinery.dm [hunk 1]
```diff
@@ -368,6 +369,11 @@ Class Procs:
 		update_icon()
 		updateUsrDialog()
 
+/obj/machinery/run_obj_armor(damage_amount, damage_type, damage_flag = 0, attack_dir)
+	if(damage_flag == "melee" && damage_amount < damage_deflection)
+		return 0
+	. = ..()
```
Comment (ShizCalev) [inline code/game/machinery/_machinery.dm]:
```suggestion
	return ..()
```

**Review by ShizCalev — DISMISSED** [review]

File: code/game/machinery/_machinery.dm [hunk 2]
```diff
@@ -368,6 +369,11 @@ Class Procs:
 		update_icon()
 		updateUsrDialog()
 
+/obj/machinery/run_obj_armor(damage_amount, damage_type, damage_flag = 0, attack_dir)
```
Comment (Cyberboss) [inline code/game/machinery/_machinery.dm]:
```suggestion
/obj/machinery/run_obj_armor(damage_amount, damage_type, damage_flag = NONE, attack_dir)
```

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by XDTM** [comment]
i think it might be worth moving it to /obj/ preemptively, i can see structures having thresholds as well

**Review by Cyberboss — APPROVED** [review]

**Comment by Cyberboss** [comment]
This is fine for now

**Comment by subject217** [comment]
as a side note, secure closets use copypaste code from this, and they don't even have the honor to make it a var. if it got moved to obj this would need to be addressed as well.
 
```/obj/structure/closet/secure_closet/run_obj_armor(damage_amount, damage_type, damage_flag = 0, attack_dir)
	if(damage_flag == "melee" && damage_amount < 20)
		return 0
. = ..()

**Pull request merged**